File: backend/app/schemas/school.py
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, HttpUrl


class SchoolBase(BaseModel):
    name: str
    school_type: Optional[str] = None
    address_street: Optional[str] = None
    address_city: Optional[str] = None
    address_postcode: Optional[str] = None
    voivodeship: Optional[str] = None
    county: Optional[str] = None
    municipality: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    website_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    is_public: Optional[bool] = None


class SchoolRead(SchoolBase):
    id: uuid.UUID
    rspo_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SchoolProfileRead(BaseModel):
    id: uuid.UUID
    school_id: uuid.UUID
    class_profiles: Optional[dict] = None
    languages_offered: Optional[list] = None
    extracurricular_activities: Optional[list] = None
    notable_achievements: Optional[list] = None
    description_summary: Optional[str] = None
    extracted_at: datetime
    model_used: Optional[str] = None

    model_config = {"from_attributes": True}


class SchoolWithProfile(SchoolRead):
    profile: Optional[SchoolProfileRead] = None


class SchoolListResponse(BaseModel):
    items: list[SchoolRead]
    total: int
    page: int
    size: int

File: frontend/src/types/school.ts
export interface School {
  id: string;
  rspo_id: string | null;
  name: string;
  school_type: string | null;
  address_street: string | null;
  address_city: string | null;
  address_postcode: string | null;
  voivodeship: string | null;
  county: string | null;
  municipality: string | null;
  latitude: number | null;
  longitude: number | null;
  website_url: string | null;
  phone: string | null;
  email: string | null;
  is_public: boolean | null;
  created_at: string;
  updated_at: string;
  profile?: SchoolProfile | null;
}

export interface SchoolProfile {
  id: string;
  school_id: string;
  class_profiles: ClassProfile[] | null;
  languages_offered: string[] | null;
  extracurricular_activities: string[] | null;
  notable_achievements: string[] | null;
  description_summary: string | null;
  extracted_at: string;
  model_used: string | null;
}

export interface ClassProfile {
  name: string;
  description?: string;
  languages?: string[];
}

export interface SchoolListResponse {
  items: School[];
  total: number;
  page: number;
  size: number;
}
